### ISSUE TITLE
[DR] Unify gradient-based closure and neural network closure infrastructure

### ISSUE THREAD

**Issue by JunoRavin** [ammarhakim/gkylzero#1042]
Currently we support two different improvements upon the traditional local closure for the 10 moment system of equations

<img width="234" height="49" alt="Image" src="https://github.com/user-attachments/assets/f42bbd01-157c-4ebc-a10d-45fec98668c3" />

The first is a gradient-based closure: 
<img width="166" height="48" alt="Image" src="https://github.com/user-attachments/assets/0694632a-3a1b-4538-87ab-cbddb8ebc8a4" />

where the bracket is the symmetrization of the rank-3 tensor from the partial derivatives of the temperature tensor. The second is a neural network closure in which we utilize a kinetic simulation (typically the PKPM model) to learn various components of the heat flux from lower moments $`\mathbf{Q}_{\boldsymbol{\theta}}`$ where $`\boldsymbol{\theta} = (\rho, p_\parallel, p_\perp, \nabla \rho, \dots)`$ are the input parameters to the model. 

In the current version of the code, @kltrbrds nicely modularized the gradient-based closure in a private header file: https://github.com/ammarhakim/gkeyll/blob/main/moments/zero/gkyl_ten_moment_grad_closure_priv.h

where there are separate functions for getting the heat flux vector at vertices: https://github.com/ammarhakim/gkeyll/blob/main/moments/zero/gkyl_ten_moment_grad_closure_priv.h#L154

followed by functions for the divergence of the heat flux that is used to perform the final update: https://github.com/ammarhakim/gkeyll/blob/main/moments/zero/gkyl_ten_moment_grad_closure_priv.h#L200

Right now, the neural network closure converts finite volume data to DG data, and feeds this into the neural network. **My proposal**: shift the neural network interpolation to follow precisely the structure of the gradient-based closure and instead learn the heat flux as a function of these input parameters so that we can reuse the existing gradient-based closure infrastructure: 
1. Compute the density, parallel temperature, perpendicular temperature, and various gradients at the cell vertices
2. Infer the parallel and perpendicular heat flux at these points from the neural network
3. Reconstruct Q_{ijk} from the symmetrization of these parallel and perpendicular heat fluxes
4. Call the same heat flux divergence functions that the gradient-based closure utilizes
5. Update the pressure tensor from this divergence

I am going to experiment with the training procedure utilizing nodal information (the Gauss-Legendre nodal basis) so that it can still effectively utilize the higher order information in our DG PKPM representation, but such that the output vector is just two scalars regardless of polynomial order or dimensionality.

**Comment by ammarhakim** [comment]
I would like this discuss once I am back. Pick a day in the week of June 15th. We need to get both this soon, specially if we are to do complex moment simulation for mirrors, etc.
